tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/ITmfConfigurationSource.java

```diff
@@ -59,6 +87,33 @@ public interface ITmfConfigurationSource {
      */
     ITmfConfiguration update(String id, Map<String, Object> parameters) throws TmfConfigurationException;
 
+    /**
+     * Updates a configuration instance.
+     * <p>
+     * The parameters to be provided are described by
+     * {@link ITmfConfigurationSourceType#getConfigParamDescriptors()}.
+     *
+     * @param id
+     *            The configuration ID of the configuration to update
+     * @param configuration
+     *            The input configuration parameters to update a configuration instance.
+     * @return a new {@link ITmfConfiguration} if successful
+     * @throws TmfConfigurationException
+     *             If the update of the configuration fails
+     * @since 9.5
+     */
+    @SuppressWarnings("null")
+    default ITmfConfiguration update(String id, ITmfConfiguration configuration) throws TmfConfigurationException {
+        Map<String, Object> map = new HashMap<>();
+        try {
+            String json = new Gson().toJson(configuration.getParameters());
+            map = new Gson().fromJson(json, Map.class);
+        } catch (Exception e) {
+            throw new TmfConfigurationException("Can't convert input configuration to Map to update configuration", e); //$NON-NLS-1$
+        }
+        return update(id, map);
```

Comment:
You can just pass configuration.getParameters() directly?

Reply:
True. That will make it simpler and not dependency to GSON is added.